iothub_client/samples/iothub_client_device_twin_and_methods_sample/iothub_client_device_twin_and_methods_sample.c

```diff
@@ -214,168 +580,234 @@ static int deviceMethodCallback(const char* method_name, const unsigned char* pa
         result = -1;
     }
 
+    (void)printf("\n"); // Formatting
+
     return result;
 }
 
-static void getCompleteDeviceTwinOnDemandCallback(DEVICE_TWIN_UPDATE_STATE update_state, const unsigned char* payLoad, size_t size, void* userContextCallback)
+//
+// Encoding/Decoding with JSON library
+//
+static bool parse_twin_document(const unsigned char* payload, CAR* out_car, int64_t* out_parsed_desired_version)
 {
-    (void)update_state;
-    (void)userContextCallback;
-    printf("GetTwinAsync result:\r\n%.*s\r\n", (int)size, payLoad);
+    bool parsed = parse_twin_document_desired(payload, out_car);
+
+    // For a GET full twin request, each section's $version may need to be parsed.
+    if (out_parsed_desired_version)
+    {
```

Comment:
does it matter that this is run unconditionally of `parsed`

Reply:
?  i'm not clear on what you mean.  This part is only run if an Lvalue is supplied, implying it is wanting to parse for the version.  Otherwise if NULL is passed, version will not be parsed for.

I'm don't think its good to parse the payload twice, which is what happens if the version is needed.  But it would only happen when the full GET is requested (one time in the sample).  Mostly I am trying to reduce code duplication for the sake of the sample.